policy/modules/system/userdomain.if

```diff
@@ -679,6 +679,10 @@ template(`userdom_common_user_template',`
 		')
 	')
 
+	ifdef(`init_systemd',`
+		systemd_role_template($1, $1_r, $1_t)
```

Comment:
Did you try putting this in the `dbus_system_bus_client` block to avoid making dbus optional in `systemd.if`? I don't think this can work w/o dbus.

Also, please put a comment for why it is out of order
1. when we forget about this we don't inadvertently put it back
2. when RHEL7 finally goes EOL we can restore it to the original location

Reply:
I hadn't tried this in dbus_system_bus_client block (until you suggested it).  But, I get the same error about duplicate definition when building.  It seems like putting systemd_role_template in an optional is causing some problem, but I really don't understand why.  

It seems to also be resolved by commenting out the use of 'systemd_user_daemon_domain' in 'dbus_role_template' which is what is requiring the 'auditadm_systemd_t'.  But doing that would change functionality.  

I will add a comment and update the pull request.  
If there is anything else you would like me to try, let me know.

Reply:
Please provide the error you're getting.

Reply:
I'm sorry @pebenito , are you asking me for the error seen when moving 'systemd_role_template' into the 'dbus_system_bus_client' optional block?  It is the same:

/usr/bin/checkmodule -m tmp/auditadm.tmp -o tmp/auditadm.mod
/usr/bin/checkmodule:  loading policy configuration from tmp/auditadm.tmp
policy/modules/roles/auditadm.te:42:ERROR 'duplicate declaration of type/attribute' at token 'auditadm_systemd_t' on line 32039:
#line 42
		type auditadm_systemd_t;
/usr/bin/checkmodule:  error(s) encountered while parsing configuration
make: *** [tmp/auditadm.mod] Error 1


If you are asking for it to be clearer in the comment in systemd.if, I can do that also.

Reply:
Unless there is some distro other than RHEL7 that hits this, I'm afraid I have to reject this piece of the PR. RHEL7 is not officially supported by refpolicy, and this is over the line of what I feel comfortable carrying. If it was just the change in `auditadm.te`, it would be fine, but I draw the line at making dbus optional when it's not.

I still do want the other pieces.